src/connector/src/source/util.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{BTreeMap, HashSet};
+use std::sync::Arc;
+
+use crate::error::{ConnectorError, ConnectorResult};
+use crate::source::nats::split::NatsSplit;
+use crate::source::SplitImpl;
+
+pub fn fill_adaptive_split(
+    discovered_splits: BTreeMap<Arc<str>, SplitImpl>,
+    actor_in_use: &HashSet<u32>,
+) -> ConnectorResult<BTreeMap<Arc<str>, SplitImpl>> {
+    debug_assert!(discovered_splits.len() == 1);
+    let split = discovered_splits.values().next().unwrap();
```

Comment:
In that case, please just pass one split in.

```suggestion
pub fn fill_adaptive_split(
    discovered_split: SplitImpl,
    actor_in_use: &HashSet<u32>,
) -> ConnectorResult<BTreeMap<Arc<str>, SplitImpl>> {
```